projects/igniteui-angular/src/lib/grids/api.service.ts

```diff
@@ -585,6 +585,12 @@ export class GridBaseAPIService <T extends IgxGridBaseComponent & IGridDataBinda
         return column.dataType === DataType.Number;
     }
 
+    public get_data(id: string): any[] {
```

Comment:
This is pretty much the same as `get_all_data` bellow. 
If this is specific to the Tree Grid shouldn't it be handled in the IgxTreeGridAPIService?

Reply:
It is in the base api service because it is used in the grid-base class. The idea is that get_all_data returns the child rows as well. However in the case of treeGrid it uses flatData which is not populated before the pipes are processed. For the auto-generate we don't need the child rows that's why we can use the grid.data value, but since the grid-base doesn't have data property, I have added this method to the service.

Reply:
I mentioned the 'get_all_data' because in the context of the other grids it already does the same thing.
Is there are a reason for the override of the `get_all_data` in the Tree Grid to return always flatData or empty array even when it is not populated yet and data is set?

Reply:
We could do that, but the idea of get_all_data is to return the all of the data including the child rows, that's why I prefer to add a separate method just for the grid.data.

Reply:
Yep and the grid.data initially also has the child rows, but it is not flattened. That's why I think it might be better to be handled by the tree grid to avoid confusion.

At least in order to avoid duplicate code the 'get_all_data' can use the 'get_data' in the base class if you decide to keep the current logic.